workspaces/core/src/search/private/components/Input.tsx

```diff
@@ -1,95 +1,81 @@
-import React, { forwardRef, Fragment, Ref } from 'react';
+import React, { ReactNode, forwardRef, Ref, useContext, LegacyRef } from 'react';
+import mergeClassnames from '../../../mergeClassnames/mergeClassnames';
 import GenericSearch from '../../../private/icons/GenericSearch';
+import { SearchContext } from '../utils/context';
 
-type SearchProps = {
-  onChange: (value: string) => void;
-  onFocus?: () => void;
-  onBlur?: (e: React.FocusEvent<HTMLElement>) => void;
-  placeholder?: string;
-  prefix?: string[];
-  value: string;
-  clear: string | React.ReactNode;
-  autoFocus?: boolean;
-};
+const Wrapper = ({ children }: { children: ReactNode }) => {
+  return <div className="flex items-center space-x-1.5 pl-3 moon-search-input">
+    {children}
+  </div>
+}
 
-function Input(
-  {
-    onChange,
-    onFocus,
-    onBlur,
-    placeholder,
-    prefix,
-    value,
-    clear,
-    autoFocus,
-  }: SearchProps,
-  ref: Ref<HTMLInputElement>
-) {
-  return (
-    <div className="flex items-center space-x-1.5 pl-3 moon-search-input">
-      <GenericSearch className="w-6 h-6 pointer-events-none text-bulma" />
+const Icon = ({ className }: { className?: string }) => {
+  return <GenericSearch
+    className={
+      mergeClassnames("w-6 h-6 pointer-events-none text-bulma", className)
+    } />
+}
 
-      {prefix?.length
-        ? prefix.map((p) => {
-            return (
-              <Fragment key={p}>
-                <span className="dark:text-bulma">{p}</span>
-                <span className="text-bulma">/</span>
-              </Fragment>
-            );
-          })
-        : null}
+const InnerInput = forwardRef(({
+  placeholder,
+  autoFocus,
+  className
+}: {
+  placeholder?: string;
```

Comment:
placeholder also can be JSX.Element  as well, please put 
 ` placeholder?: JSX.Element | string;`

Reply:
It's HTML input placeholder, it's a string in React. Other design systems (e.g MUI, Ant) pass string there